### PR TITLE
Fix backend name for Stateful services

### PR DESCRIPTION
**Pull request by ldez**
Fix backend name for Stateful services:
The name of the backend is `{{$backendName := (print $service.Name $partition.PartitionInformation.ID)}}` but the backend name in frontends is `backendName = {{$service.Name}}/{{$partitionId}}` (note the `/`)

Also:
- update Træfik version.
- Use new methods from Træfik for labels.
- remove unused methods
- rename functions in the template

### CODE REVIEW

**Comment by lawrencegripper** [comment]
Thanks, will try and take a look asap - most likely Friday, sorry for the delay

**Comment by lawrencegripper** [comment]
@petertiedemann Wanted to flag that this change will require you to refactor the template we setup as the function names exposed have been renamed. Included you here so you can see the renames required. Let me know if you have any issues with the changes.

File: labels.go [hunk 1]
```diff
@@ -1,23 +1,50 @@
...
+)
 
-func hasServiceLabel(service ServiceItemExtended, key string) bool {
-	_, exists := service.Labels[key]
-	return exists
+// Must be replace by:
+// func hasFunc(labelName string) func(service ServiceItemExtended) bool {
+// 	return func(service ServiceItemExtended) bool {
+// 		return label.Has(service.Labels, labelName)
+// 	}
```
Comment (lawrencegripper) [inline labels.go]: I'll move to this model with the work on #3. As discussed there I'm guessing we're ok to keep the generic functions to allow easy customization in the templates, such as adding custom labels or changing behaviors?

Comment (ldez) [reply labels.go]: I agree but we also need to remove some "logical" from template.
Example:

before:
```
weight = {{getLabelValueWithDefault $service "backend.group.weight" "1"}}
```
after:
```
weight = {{ getWeight $service }}
```

**Comment by ldez** [comment]
I have done really bad things:
- too many things in this PR
- mix a fix for `1.5` and a refactor related to `master`

I will revert and redo this PR (in 2 PR this time)
Sorry for this.

**Pull request merged**